Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/QdmModelInfoProvider.java

```diff
@@ -8,8 +8,30 @@
  * Created by Bryn on 2/3/2016.
  */
 public class QdmModelInfoProvider implements ModelInfoProvider {
+    private String version;
+    public String getVersion() {
+        return version;
+    }
+    public void setVersion(String version) {
+        this.version = version;
+    }
+    public QdmModelInfoProvider withVersion(String version) {
+        setVersion(version);
+        return this;
+    }
+
     public ModelInfo load() {
-        return JAXB.unmarshal(QuickModelInfoProvider.class.getResourceAsStream("/gov/healthit/qdm/qdm-modelinfo-5.0.xml"),
-                ModelInfo.class);
+        switch (version) {
+            case "4.1.2":
+                return JAXB.unmarshal(QdmModelInfoProvider.class.getResourceAsStream("/gov/healthit/qdm/qdm-modelinfo.xml"),
+                        ModelInfo.class);
+            case "4.2":
+                return JAXB.unmarshal(QuickModelInfoProvider.class.getResourceAsStream("/gov/healthit/qdm/qdm-modelinfo-4.2.xml"),
+                        ModelInfo.class);
+            case "5.0":
+            default:
```

Comment:
The intent was to allow 5.0 to be the default when no version is specified, but you make a good point, this would also change invalid versions to be 5.0 as well. I'll make the change, thanks!